### PR TITLE
[IMP] orm: name_get into display_name

### PR DESCRIPTION
**Pull request by ryv-odoo**
https://github.com/odoo/odoo/pull/122085
https://github.com/odoo/enterprise/pull/42599
https://github.com/odoo/upgrade/pull/4780

### CODE REVIEW

File: content/developer/reference/backend/orm.rst [hunk 1]
```diff
@@ -407,6 +407,13 @@ Automatic fields
 
     Raise an Error otherwise.
 
+.. attribute:: Model.display_name
+
+    Name :class:`field <odoo.fields.String>` displayed by default in the web client
+
+    By default, it equals to :attr:`~odoo.models.BaseModel._rec_name` value field
+    but the behavior can be customize by overriding :attr:`~odoo.models.BaseModel._compute_display_name`
```
Comment (Feyensv) [inline content/developer/reference/backend/orm.rst:415]:
```suggestion
    but the behavior can be customized by overriding :attr:`~odoo.models.BaseModel._compute_display_name`
```

File: content/developer/reference/backend/orm/changelog.rst [hunk 2]
```diff
@@ -4,6 +4,12 @@
 Changelog
 =========
 
+Odoo Online version 16.4
+========================
+
+- `odoo.models.Model.name_get` has been deprecated with
+  TODO. Instead read display_name.
```
[the remark this answers is not in the document]
Comment (AntoineVDV) [reply content/developer/reference/backend/orm/changelog.rst:11]:
```suggestion
  TODO. Read `display_name` instead.
```

File: content/developer/reference/backend/orm.rst [hunk 3]
```diff
@@ -407,6 +407,13 @@ Automatic fields
 
     Raise an Error otherwise.
 
+.. attribute:: Model.display_name
+
+    Name :class:`field <odoo.fields.String>` displayed by default in the web client
```
Comment (Feyensv) [inline content/developer/reference/backend/orm.rst:412]: Isn't the display_name a `Char` ?
```python
            add_default('display_name', fields.Char(
                string='Display Name', automatic=True, compute='_compute_display_name'))
```

(thought i added that comment to my review but it seems i didn't :man_facepalming:)

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Review by Julien00859 — APPROVED** [review]

**Review by VincentSchippefilt — APPROVED** [review]

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Comment by ryv-odoo** [comment]
@robodoo r+

**Pull request closed without merging**